### ISSUE TITLE
Trying to access an object attribute on the class itself doesn't raise errors

### ISSUE THREAD

**Issue by tusharsadhwani** [python/mypy#15307]
**To Reproduce**

```python
class SimpleClass:
    def __init__(self) -> None:
        self.a = "test"

print(SimpleClass.a)
```

**Expected Behavior**

I was expecting `SimpleClass.a` to be caught as an error.

Pyright does catch it:

```
asd.py
  asd.py:5:19 - error: Cannot access member "a" for type "Type[SimpleClass]"
    Member "a" is unknown (reportGeneralTypeIssues)
```

**Actual Behavior**

No issues found.

**Your Environment**

- Mypy version used: 1.3.0
- Mypy command-line flags: `--strict`
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.11

**Comment by hauntsaninja** [comment]
Duplicate of #240
